presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -231,8 +234,9 @@ private static ParquetPageSource createParquetPageSource(
                 prestoTypes.add(column.getType());
 
                 internalFields.add(parquetField.flatMap(field -> {
-                    String columnName = useParquetColumnNames ? column.getName() : fileSchema.getFields().get(column.getBaseHiveColumnIndex()).getName();
-                    return constructField(column.getType(), lookupColumnByName(messageColumnIO, columnName));
+                    String columnName = useParquetColumnNames ? column.getBaseColumnName() : fileSchema.getFields().get(column.getBaseHiveColumnIndex()).getName();
+                    ColumnRef columnRef = resolveNestedColumn(new ColumnRef(column.getBaseType(), lookupColumnByName(messageColumnIO, columnName)), column.getHiveColumnProjectionInfo());
+                    return constructField(columnRef.getType(), columnRef.getColumn());
```

Comment:
This will create a PrimitiveField using the nested column path from the column.getHiveColumnProjectionInfo(). This seems to work except in some cases - see testIsNotNullWithNestedData for example. This test references a nested column a.y when the row a is null.

Should the field constructed here include the RowType(s) here? I've played around with that, but end up getting an UnsupportedOperationException when an attempt is made to read the data. However, I can get the testIsNotNullWithNestedData to pass.